### PR TITLE
Apply clang format style from Cabana

### PR DESCRIPTION
**Pull request by aetx**
Just applied the clang format style from Cabana.

### CODE REVIEW

**Review by streeve — COMMENTED** [review]
Can you copy the `cmake/FindCLANG_FORMAT.cmake` from Cabana and add to CMake?

**Comment by aetx** [comment]
Done. Although when running `make format` I get the error `error: cannot use -i when reading from stdin`. Is there something else to consider than just adding the package and target to the main `CMakeLists.txt`? I do not get the error in Cabana.

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,8 +16,21 @@ if( NOT Cabana_ENABLE_CAJITA )
...
 add_subdirectory(src)
 
 # examples
 add_subdirectory(examples)
+
+##---------------------------------------------------------------------------##
+## Clang Format
+##---------------------------------------------------------------------------##
+if(CLANG_FORMAT_FOUND)
+  file(GLOB_RECURSE FORMAT_SOURCES core/*.cpp core/*.hpp cajita/*hpp cajita/*cpp example/*cpp example/*hpp cmake/*cpp cmake/*hpp benchmark/*cpp benchmark/*hpp)
```
Comment (streeve) [inline CMakeLists.txt]:
```suggestion
  file(GLOB_RECURSE FORMAT_SOURCES src/*.cpp src/*.hpp examples/*cpp examples/*hpp)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,8 +16,21 @@ if( NOT Cabana_ENABLE_CAJITA )
...
 add_subdirectory(src)
 
 # examples
 add_subdirectory(examples)
+
+##---------------------------------------------------------------------------##
+## Clang Format
+##---------------------------------------------------------------------------##
+if(CLANG_FORMAT_FOUND)
+  file(GLOB_RECURSE FORMAT_SOURCES src/*.cpp src/*.hpp examples/*cpp examples/*hpp)
```
Comment (junghans) [inline CMakeLists.txt]:
```suggestion
  file(GLOB_RECURSE FORMAT_SOURCES src/*.cpp src/*.hpp examples/*.cpp examples/*.hpp)
```

**Review by junghans — CHANGES_REQUESTED** [review]

**Comment by streeve** [comment]
@junghans is it finding a newer `clang-format`?

**Comment by junghans** [comment]
> @junghans is it finding a newer `clang-format`?

Yes, any `clang-format` can be used, only if it is called `clang-format-XX` (XX>10) then is won't pick it up.

**Comment by junghans** [comment]
`Could NOT find CLANG_FORMAT: Found unsuitable version "0.0", but required is at least "10" (found /usr/bin/clang-format)`
Not sure why it detected `0.0`.

**Review by junghans — APPROVED** [review]

**Review by streeve — APPROVED** [review]

**Comment by junghans** [comment]
ping @sslattery your call!

**Comment by junghans** [comment]
@streeve can port 34805d03a0f317fef5ff36764f556b2661e89d25 back to Cabana as well?

**Comment by junghans** [comment]
@streeve `git am` is magic: https://github.com/ECP-copa/Cabana/pull/440

**Pull request merged**